py/dml/ctree.py

```diff
@@ -595,6 +596,27 @@ def mkInlinedMethod(site, method, pre, body, post):
             collapsed.append(stmt)
     return InlinedMethod(site, method, pre, collapsed, post)
 
+class IndicesAssert(Statement):
+    @auto_init
+    def __init__(self, site, method, indices): pass
+
+    def toc_stmt(self):
+        logname_indices = self.method.logname_anonymized(
+            ('%u',) * self.method.dimensions)
+        logname_dimsizes = self.method.parent.logname_anonymized(
+            self.method.dimsizes)
+        fmt = (f"Attempt made to call {logname_indices}(), which involves "
+               + "one or more out-of-bounds array accesses. Array dimensions "
+               + f"are as follows: {logname_dimsizes}")
+        check = ' && '.join(f'({idx.read()}) < {size}'
+                            for (idx, size) in zip(self.indices,
+                                                   self.method.dimsizes))
+        self.linemark()
+        idx_args = ', '.join(f'({idx.read()})' for idx in self.indices)
+        out(f'ASSERT_FMT({check}, "{fmt}", {idx_args});\n')
```

Comment:
`FATAL_ERROR_IF` is preferred over `ASSERT_FMT`. Also, I'm a bit worried about increased code expansion; the message is duplicated many times. Perhaps the check should be abstracted into a separate C function in dmllib.h.

Reply:
Perhaps a message based on (object-identity, dml-filename, dml-line, variable-name) is sufficient?

Reply:
Replaced this with an mechanism that should tick all boxes.